WireAPI/Sources/WireAPI/APIs/TeamsAPI/TeamsAPIV0.swift

```diff
@@ -0,0 +1,289 @@
+            .parse(response)
+    }
+
+    // MARK: - Get team members
+
+    func getTeamMembers(
+        for teamID: Team.ID,
+        maxResults: UInt
+    ) async throws -> [TeamMember] {
+        var components = URLComponents(string: "\(basePath(for: teamID))/members")
+        components?.queryItems = [URLQueryItem(name: "maxResults", value: "2000")]
+
+        guard let path = components?.url?.absoluteString else {
+            throw TeamsAPIError.failedToGenerateRequest
+        }
+
+        let request = HTTPRequest(
+            path: path,
+            method: .get
+        )
+
+        let response = try await httpClient.executeRequest(request)
+
+        return try ResponseParser()
+            .success(code: 200, type: TeamMemberListResponseV0.self)
+            .failure(code: 400, error: TeamsAPIError.invalidQueryParmeter)
+            .failure(code: 403, label: "no-team-member", error: TeamsAPIError.selfUserIsNotTeamMember)
+            .failure(code: 404, error: TeamsAPIError.teamNotFound)
+            .parse(response)
+    }
+
+    // MARK: - Get legalhold status
+
+    func getLegalholdStatus(
+        for teamID: Team.ID,
+        userID: UUID
+    ) async throws -> LegalholdStatus {
+        let request = HTTPRequest(
+            path: "\(basePath(for: teamID))/legalhold/\(userID.transportString())",
+            method: .get
+        )
+
+        let response = try await httpClient.executeRequest(request)
+
+        return try ResponseParser()
+            .success(code: 200, type: LegalholdStatusResponseV0.self)
+            .failure(code: 404, error: TeamsAPIError.invalidRequest)
+            .failure(code: 404, label: "no-team-member", error: TeamsAPIError.teamMemberNotFound)
+            .parse(response)
+    }
+
+}
+
+struct TeamResponseV0: Decodable, ToAPIModelConvertible {
+
+    let id: UUID
+    let name: String
+    let creator: UUID
+    let icon: String
+    let icon_key: String?
+    let binding: Bool?
+
+    func toAPIModel() -> Team {
+        Team(
+            id: id,
+            name: name,
+            creatorID: creator,
+            logoID: icon,
+            logoKey: icon_key,
+            splashScreenID: nil
+        )
+    }
+
+}
+
+struct ConversationRolesListResponseV0: Decodable, ToAPIModelConvertible {
+
+    let conversation_roles: [ConversationRoleResponseV0]
+
+    func toAPIModel() -> [ConversationRole] {
+        conversation_roles.map { $0.toAPIModel() }
+    }
+
+}
+
+struct ConversationRoleResponseV0: Decodable {
+
+    let conversation_role: String?
+    let actions: [ConversationActionResponseV0]
+
+    func toAPIModel() -> ConversationRole {
+        ConversationRole(
+            name: conversation_role ?? "unknown",
+            actions: Set(actions.map {
+                $0.toAPIModel()
+            })
+        )
+    }
+
+}
+
+enum ConversationActionResponseV0: String, Decodable {
+
+    case add_conversation_member
```

Comment:
Same question: should we use snake case for vars here?

Reply:
done 77f2f3fad0fd7d002b837cc5caddd332ebc3f9a3